cryptoki/tests/basic.rs

```diff
@@ -781,3 +781,64 @@ fn ro_rw_session_test() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+#[serial]
+fn aes_cbc_encrypt() -> Result<()> {
+    // Encrypt two blocks of zeros with AES-128-CBC, and zero IV
+    let key = vec![0; 16];
+    let iv = [0; 16];
+    let plain = [0; 32];
+    let expected_cipher = [
+        0x66, 0xe9, 0x4b, 0xd4, 0xef, 0x8a, 0x2c, 0x3b, 0x88, 0x4c, 0xfa, 0x59, 0xca, 0x34, 0x2b,
+        0x2e, 0xf7, 0x95, 0xbd, 0x4a, 0x52, 0xe2, 0x9e, 0xd7, 0x13, 0xd3, 0x13, 0xfa, 0x20, 0xe9,
+        0x8d, 0xbc,
+    ];
+
+    let (pkcs11, slot) = init_pins();
+    let session = pkcs11.open_rw_session(slot)?;
+    session.login(UserType::User, Some(USER_PIN))?;
+
+    let template = [
+        Attribute::Class(ObjectClass::SECRET_KEY),
+        Attribute::KeyType(KeyType::AES),
+        Attribute::Value(key),
+        Attribute::Encrypt(true),
+    ];
+    let key_handle = session.create_object(&template)?;
+    let mechanism = Mechanism::AesCbc(iv);
+    let cipher = session.encrypt(&mechanism, key_handle, &plain)?;
+    assert_eq!(expected_cipher[..], cipher[..]);
+    Ok(())
```

Comment:
Looks OK but I've been wondering if you could stuff in `decrypt` too to see if the decrypted plaintext matches the input?

Reply:
I'm ok either way on this, tbh, don't know if we need to test both directions since I assume we're testing the conversion of the mechanism here.

Reply:
Yeah, sorry, maybe I got carried over since the test already checks for expected ciphertext so let's not inflate the test code artificially here.